storage/merkle/tree.go

```diff
@@ -273,6 +268,99 @@ GetLoop:
 	}
 }
 
+// Prove constructs inclusion proof for a given key if the key exists in the trie.
+// It traverses the trie from top to down and collects data for proof as follows:
+//  - if full node, capture the sibling node hash value and append zero to short counts
+//  - if short node, appends the node.shortCount to the short count list
+//  - if leaf, would capture the leaf value
+func (t *Tree) Prove(key []byte) (*Proof, bool) {
```

Comment:
A brief documentation of the return values would be great. Thanks

Reply:
added